x/interchainqueries/types/tx.go

```diff
@@ -13,7 +21,30 @@ func (msg MsgSubmitQueryResult) Type() string {
 }
 
 func (msg MsgSubmitQueryResult) ValidateBasic() error {
-	// TODO: add basic validation.
+	if msg.Result == nil {
+		return sdkerrors.Wrap(ErrEmptyResult, "query result can't be empty")
+	}
+
+	if msg.Result.KvResults == nil && msg.Result.Blocks == nil {
```

Comment:
I think it might be a good idea to actually check that we have some values in these arrays:
```suggestion
	if len(msg.Result.KvResults) == 0 && len(msg.Result.Blocks) == 0 {
```